src/test/ui/consts/const-eval/ub-enum-overwrite.rs

```diff
@@ -0,0 +1,18 @@
+#![feature(const_mut_refs)]
+#![allow(const_err)] // make sure we cannot allow away the errors tested here
```

Comment:
It is pure coincidence that this error only shows up during const validation and not during evaluation, and as such is outside `const_err`. And anyway even if `const_err` is allowed, it is still a hard error to *use* any constant that caused such an error.

So, I see no good reason to `allow` that lint here. If the UB triggered that lint, that would still be fine.